src/docs/resources/architectural-overview.md

```diff
@@ -0,0 +1,966 @@
+layout](https://api.flutter.dev/flutter/widgets/Row-class.html) of some
+[text](https://api.flutter.dev/flutter/widgets/Text-class.html) and
+[various](https://api.flutter.dev/flutter/material/IconButton-class.html)
+[buttons](https://api.flutter.dev/flutter/material/PopupMenuButton-class.html).
+As needed, the framework recursively asks each widget to build until the tree is
+entirely described by [concrete renderable
+objects](https://api.flutter.dev/flutter/widgets/RenderObjectWidget-class.html).
+The framework then stitches together the renderable objects into a renderable
+object tree.
+
+A widget’s build function should be free of side effects. Whenever the function
+is asked to build, the widget should return a new tree of widgets<sup><a
+href="#a1">1</a></sup>, regardless of what the widget previously returned. The
+framework does the heavy lifting work to determine which build methods need to
+be called based on the render object tree (described in more detail later). More
+information about this process can be found in the [Inside Flutter
+topic](https://flutter.dev/docs/resources/inside-flutter#linear-reconciliation).
+
+On each rendered frame, Flutter can recreate just the parts of the UI where the
+state has changed by calling that widget’s `build()` method. Therefore it is
+important that build methods should return quickly, and heavy computational work
+should be done in some asynchronous manner and then stored as part of the state
+to be used by a build method.
+
+While relatively naïve in approach, this automated comparison is quite
+effective, enabling high-performance, interactive apps. And, the design of the
+build function simplifies your code by focusing on declaring what a widget is
+made of, rather than the complexities of updating the user interface from one
+state to another.
+
+### Widget state
+
+The framework introduces two major classes of widget: _stateful_ and _stateless_
+widgets.
+
+Many widgets have no mutable state: they don’t have any properties that change
+over time (for example, an icon or a label). These widgets subclass
+[`StatelessWidget`](https://api.flutter.dev/flutter/widgets/StatelessWidget-class.html).
+
+However, if the unique characteristics of a widget need to change based on user
+interaction or other factors, that widget is _stateful_. For example, if a
+widget has a counter that increments whenever the user taps a button, then the
+value of the counter is the state for that widget. When that value changes, the
+widget needs to be rebuilt to update its part of the UI. These widgets subclass
+[`StatefulWidget`](https://api.flutter.dev/flutter/widgets/StatefulWidget-class.html), and (because the widget itself is immutable) they store mutable state in a separate class that subclasses [`State`](https://api.flutter.dev/flutter/widgets/State-class.html). `StatefulWidget`s don’t have a build method; instead, their user interface is built through their `State` object.
```

Comment:
Such a good point, and I don't think I say this anywhere else.

Btw, this sentence is WAY too long. Our convention is 80 chars or less. We often break on clauses.

Reply:
Somehow my wrapping missed this. Fixed.